File: lab02/create.py
import element as e

class Create(e.Element):
    def __init__(self, delay):
        super().__init__(delay)

    # коли пристрій вільний - збільшуємо лічильник кількості
    def out_act(self):
        super().out_act()
        self.t_next[0] = self.t_curr + self.get_delay()
        self.next_element[0].in_act()

File: lab02/src/Create.java
import java.util.List;

public class Create extends Event{
    public Create(double delay, String name) {
        super(delay, name);
    }
    @Override
    public void outAct(double currentTime, List<Event> events) {
        super.outAct(currentTime, events);
        eventTime = currentTime + getDelay();
        next.inAct(currentTime);
    }
    @Override
    public void doStatistics(double delta) {
        meanQueue += queue * delta;
    }
}
